src/client/validation.ts

```diff
@@ -81,6 +82,28 @@ export function parseAppChannelName(name: string): ChannelId {
     return name;
 }
 
+/**
+ * Validates the provided number is an integer
+ */
+export function parseInteger(number: number): number {
+    if (typeof number !== 'number' || isNaN(number) || Math.floor(number) !== number) {
+        throw new TypeError(`${safeStringify(number, 'The provided value')} is not a valid integer`);
+    }
+
+    return number;
+}
+
+/**
+ * Validates the provided app directory data - can be either a URL or an array of Applications
+ */
+export function parseAppDirectoryData(data: string | Application[]): string | Application[] {
```

Comment:
I'd suffix these functions with `is` or `validate` instead of `parse`.

Reply:
Yeah, parse isn't the right word, but wanted to communicate that these actually return something.

Have gone withe `sanitize`, which I think better capture "unsafe data in, safe data out"